src/worker_pool/balancer/index.ts

```diff
@@ -0,0 +1,43 @@
+import type { PiscinaTask } from '../../task_queue';
+import type { PiscinaWorker } from '..';
+
+export type PiscinaLoadBalancerCommand = {
+  candidate?: PiscinaWorker | null; // If candidate is passed, it will be used as the result of the load balancer and ingore the command
+  command?: 0 | 1; // 1 = add, 0 = busy
+};
+export type PisicnaLoadBalancer = (
+  task: PiscinaTask,
+  workers: PiscinaWorker[]
```

Comment:
Are the PiscinaWorker[] exposing their usage statistics? Without them, the load balancer code is not able to take any decision besides algorithm like round robin. And since the usage statistics are currently dispatched in different thread pool private properties, adding them to the API without refactoring will end with an quite ugly LB API.

Reply:
I was assessing wether or not to provide that detailed information; currently my goal was to provide the overall usage (max capacity, tasks running, etc.) to help the LB make decisions.

IMO it provides enough for simple algorithms like RB, but for more complex ones based on usage over time, it will require further data.

Reply:
@ronag is contacting `worker_threads` pool maintainers one after the other to ask for load balancing features :)

Since I've already spent some times on them in poolifier, without per worker usage stats, designing useful LB will be impossible.

Reply:
While thinking of it: non ready workers must be removed from the array supplied to the LB to avoid code to test for it on each LB implementation. Is `this.workers` safe regarding worker readyness?

Plenty of works for the months to come :)

Reply:
>While thinking of it: non ready workers must be removed from the array supplied to the LB to avoid code to test for it on each LB implementation. Is this.workers safe regarding worker readyness?

Partially, the `Pool` categorizes the workers between ready and non-ready, but when iterating it goes through all of them instead of just the ready ones.

Room for improvement for sure 

>Plenty of works for the months to come :)

Definitely, but plenty of good ideas and optimizations are being made here; thanks for the insightful discussion!